src/modules/brightness.rs

```diff
@@ -0,0 +1,343 @@
+                }
+            }
+            BrightnessDataSource::Login1Fs { subsystem, name } => {
+                let name = name
+                    .clone()
+                    .or_else(|| default_resource_name.clone())
+                    .expect(
+                        "Could not get resource name, consider explicit setting datasource.name",
+                    );
+                if let Err(e) = client
+                    .screen_writer()
+                    .set_brightness(subsystem.to_string(), name, brightness as u32)
+                    .await
+                {
+                    tracing::error!(?e, "Could not change brightness");
+                }
+            }
+        };
+    }
+}
+
+#[derive(Debug, Clone, Copy)]
+pub enum UiEvent {
+    AdjustBrightnessScroll(f64),
+    /// Does nothing more than force a fresh of the brightness
+    Refresh,
+}
+
+impl Module<Button> for BrightnessModule {
+    type SendMessage = BrightnessProperties;
+    type ReceiveMessage = UiEvent;
+
+    module_impl!("brightness");
+
+    fn spawn_controller(
+        &self,
+        _info: &ModuleInfo,
+        context: &WidgetContext<Self::SendMessage, Self::ReceiveMessage>,
+        mut rx: mpsc::Receiver<Self::ReceiveMessage>,
+    ) -> Result<()> {
+        let tx = context.tx.clone();
+        let ctx = context.controller_tx.clone();
+        let client = context.try_client::<brightness::Client>()?;
+        let icons = self.icons.clone();
+        let datasource = self.datasource.clone();
+        let scroll_speed = self.smooth_scroll_speed;
+        let duration = tokio::time::Duration::from_millis(self.interval);
+        let default_resource_name =
+            if let BrightnessDataSource::Login1Fs { subsystem, .. } = &datasource {
+                default_resource_name(subsystem)
+            } else {
+                None
+            };
+
+        spawn(async move {
+            // make sure we have a value on startup and not have to wait for 1 interval
+            let _ = ctx.send(UiEvent::Refresh).await;
+
+            let mut partial_scroll: f64 = 0.0;
+
+            loop {
+                let event = tokio::select! {
+                    v = rx.recv() => v,
+                    _ = sleep(duration) => None,
+                };
+
+                let (mut percent, cur, max) =
+                    match Self::read_percentage(&client, &datasource, &default_resource_name).await
+                    {
+                        Ok(d) => d,
+                        Err(e) => {
+                            tracing::error!(?e, "Could not retrieve brightness levels");
+                            continue;
+                        }
+                    };
+
+                if let Some(UiEvent::AdjustBrightnessScroll(dy)) = event {
+                    partial_scroll += dy * scroll_speed;
+
+                    if partial_scroll.abs() >= 1.0 {
+                        let no_steps = partial_scroll.floor() as i32;
```

Comment:
`no_steps` sounds more like a number related to zero steps

```suggestion
                        let num_steps = partial_scroll.floor() as i32;
```